cmd/gcp-controller-manager/app/ca_cache.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	// This should (almost) never change. If root key is compromised, rotation
+	// will be a long and painful effort.
+	rootCertURL = "https://pki.goog/cloud_integrity/tpm_ek_root_1.crt"
+	// Intermediate CA URL looks like
+	// http://pki.goog/cloud_integrity/tpm_ek_intermediate_h1_2018.crt.
+	intermediateCAPrefix = "http://pki.goog/cloud_integrity/tpm_ek_intermediate_"
+
+	crlCacheDuration = 10 * time.Minute
+)
+
+// rootCACache is a shared global cache of root CA and CRLs for TPM EK
+// certificates.
+var rootCACache = &caCache{
```

Comment:
Why does this have to be global?

Reply:
Good point, it doesn't have to be. Moved to `approverOptions`